src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,16 +1,203 @@
+            .withZone( ZoneId.systemDefault() );
+    //ZoneId.of("America/Chicago")
+    return formatter.format(creationTime);
+  }
+  /**
+   * Formats User data into readable string.
+   */
+  private Activity formatUser(User user) {
+    String creationTime = formatTime(user.getCreationTime());
+    String output = user.getName() + " joined Charmer Chat.";
+    return new Activity(creationTime,output);
+  }
+  /**
+   * Formats Conversation data into readable string.
+   */
+  private Activity formatConversation(Conversation conversation) {
+    User owner = userStore.getUserWithId(conversation.getOwnerId());
+    String creationTime = formatTime(conversation.getCreationTime());
+    String output = owner.getName() + " started conversation " + conversation.getTitle();
+    return new Activity(creationTime,output);
+  }
+  /**
+   * Formats Message data into readable string.
+   */
+  private Activity formatMessage(Message message) {
+    User author = userStore.getUserWithId(message.getAuthorId());
+    Conversation conversation = conversationStore.getConversationWithId(message.getConversationId());
+    String creationTime = formatTime(message.getCreationTime());
+    System.out.println(author);
+    System.out.println(conversation);
+    System.out.println(message);
+    String output = author.getName() + " sent in " + conversation.getTitle() + ": " + message.getContent();
+    return new Activity(creationTime,output);
+  }
+
+  /**
+   * Gets all activity on app, iterates through to sort by timestamp and prepares printed statements.
+   */
+  List<Activity> getActivities() {
+    List<Activity> activities = new LinkedList<>();
+    LinkedList<Conversation> conversations = new LinkedList<>(conversationStore.getAllConversations());
+    System.out.println(conversationStore);
+    LinkedList<User> users = new LinkedList<>(userStore.getAllUsers());
+    LinkedList<Message> messages = new LinkedList<>(messageStore.getAllMessages());
+
+    System.out.print("\n");
+    System.out.print("Length of conversations: ");
+    System.out.print(conversations.size());
+    System.out.print("\n");
+    System.out.print("Length of messages: ");
+    System.out.print(messages.size());
+    System.out.print("\n");
+    System.out.print("Length of users: ");
+    System.out.print(users.size());
+    System.out.print("\n");
+
+    while (!conversations.isEmpty() || !users.isEmpty() || !messages.isEmpty()) {
+      // Iterates through all new conversations, users and messages in the lists, compares the creation times of
+      //  each then inserts objects chronologically into list of activities.
+      ActivityType nextActivity = null;
+      Instant nextActivityTime = null;
+
+      if (!conversations.isEmpty()) {
+        nextActivity = ActivityType.ConversationCreated;
+        nextActivityTime = conversations.peek().getCreationTime();
+      }
+      if (!users.isEmpty()) {
+        if (nextActivityTime == null) {
```

Comment:
Since the null check logic is the same check as the time check logic, I might combine them into the same conditional. Something like:

```
if (nextActivityType == null || users.peek().getCreationTime().compareTo(nextActivityTime) < 0) {
...
}
```

Reply:
Sane for the other block